pkg/systemd/logsJournal.jsx

```diff
@@ -137,16 +137,8 @@ export class JournalBox extends React.Component {
                 .done(() => {
                     this.setState({ streamFinished: true });
 
-                    if (!last) {
-                        const journalctlOptions = {
-                            boot,
-                            count: 0,
-                            follow: true,
-                            grep,
-                            priority,
-                            since,
-                            until,
-                        };
+                    if (!last && !promise.stopped) {
+                        Object.assign(journalctlOptions, { since, count: 0, follow: true });
```

Comment:
These 2 added lines are not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-19041-20230629-103311-398fec5e-fedora-38-devel-expensive//Coverage/lcov/github-pr.diff.gcov.html)